WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java

```diff
@@ -145,12 +145,7 @@ private void changeLanguage(String languageCode) {
         conf.locale = newLocale;
         res.updateConfiguration(conf, res.getDisplayMetrics());
 
-        if (LanguageUtils.getCurrentDeviceLanguage(WordPress.getContext()).equals(newLocale)) {
-            // remove custom locale key when original device locale is selected
-            mSettings.edit().remove(LANGUAGE_PREF_KEY).apply();
-        } else {
-            mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
-        }
+        mSettings.edit().putString(LANGUAGE_PREF_KEY, newLocale.toString()).apply();
```

Comment:
Note: I can see that `WPActivityUtils.applyLocale()` is supposed to skip applying the locale when it's the same as the device's, but that doesn't seem to always work. I changed both the app and device locale to French, but there was a difference between the two still (see screenshot of debug code below).



<img width="721" alt="locale" src="https://cloud.githubusercontent.com/assets/3903757/21743802/e336e2dc-d4d7-11e6-808c-31838a3bddb7.png">